classification_service/main.py

```diff
@@ -59,17 +59,22 @@ def process_event(self, event_type, data):
 
         url = f"{self.model_url}/predict_{event_type}"
 
+        response = None
         with self.request_latency.time():
             try:
                 response = requests.post(
                     url,
                     data=data_json,
                     headers={"Content-Type": "application/json"},
-                    timeout=10
+                    timeout=20
                 )
             except requests.Timeout:
                 print("The request timed out")
 
+        if response is None:
```

Comment:
suggestion: should it retry the request first, with some exponential backoff, before hard failing? There could always be some intermittent network issues even if things are working fine otherwise.

Reply:
How would you recommend having it retry?

Reply:
in renku-python we have this: https://github.com/SwissDataScienceCenter/renku-python/blob/b2418a15c89e16e4442a1ecf11f703f4e15922e8/renku/core/util/requests.py#L205 specifically for `requests`.

And it's used something like:
```
with _retry(total_requests=3) as session:
    session.get(url)
```
maybe not with the `_` in the name though :slightly_smiling_face: 
```

Reply:
Should be better now!